src/packagedcode/buildpack.py

```diff
@@ -0,0 +1,143 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# ScanCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/scancode-toolkit for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import toml
+from packagedcode import models
+from packageurl import PackageURL
+
+class BuildpackHandler(models.DatafileHandler):
+    """
+    Handle buildpack.toml manifests.
+    See https://buildpacks.io/ for details on buildpack format.
+    """
+    datasource_id = "buildpack_toml"
+    path_patterns = ("*buildpack.toml",)
+    default_package_type = "buildpack"
```

Comment:
I don't think we can have a new package type here, this is not in purl-spec and possibly out of scope.
See discussions: https://github.com/package-url/purl-spec/issues/126 mentions buildpack and https://github.com/package-url/purl-spec/issues/127 has a more general discussion.
Maybe we use `generic` instead? I see `generic` being used in a dependency purl in an example below, maybe this is better.